devnet/cmd/scale-down-ecs-service/service_scaler.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/ecs/ecsiface"
+)
+
+type serviceScaler struct {
+	tickerDuration time.Duration
+	cluster        string
+	ecs            ecsiface.ECSAPI
+}
+
+func newServiceScaler(tickerDuration time.Duration, cluster string, ecs ecsiface.ECSAPI) *serviceScaler {
+	return &serviceScaler{
+		tickerDuration: tickerDuration,
+		cluster:        cluster,
+		ecs:            ecs,
+	}
+}
+
+func (ss serviceScaler) findServiceArns(ctx context.Context, serviceRegex string) (serviceArns []*string, err error) {
+	r, err := regexp.Compile(serviceRegex)
+	if err != nil {
+		return
+	}
+
+	var lsi = &ecs.ListServicesInput{
+		Cluster: &ss.cluster,
+	}
+	for {
+		var lso *ecs.ListServicesOutput
+		lso, err = ss.ecs.ListServicesWithContext(ctx, lsi)
+		if err != nil {
+			return
+		}
+		for _, arn := range lso.ServiceArns {
+			if r.MatchString(*arn) {
+				serviceArns = append(serviceArns, arn)
+			}
+		}
+		if lso.NextToken == nil {
+			break
+		}
+		lsi.NextToken = lso.NextToken
+	}
+
+	if len(serviceArns) == 0 {
+		err = fmt.Errorf("unable to locate any services with query: %s", serviceRegex)
+	}
+	return
```

Comment:
I understand you like naked returns but this is rather confusing.

It's much easier to read an early return and clothed return like

```suggestion
		return nil, fmt.Errorf("unable to locate any services with query: %s", serviceRegex)
	}
	return serviceArns, nil
```

I don't want to push my opinion too hard either, but this is really hurting readability I think. You need to keep looking back up at the signature of the function to know what's auto-returned.

Reply:
y'all really hating on naked returns.  I find it pretty explicit that we require named return values to even use naked returns.  And the function is only 30 lines long, so I don't think it's that much of an issue.  I would agree with you if the function was 50 lines, and I believe there's a linter to ensure no naked returns after a certain line length.

Reply:
:smile: not *all* either! I guess I would not use them ever, but it's supported by the language so up to you! I will complain internally (in my localhost brain) in the future haha